src/main/java/com/houarizegai/calculator/Calculator.java

```diff
@@ -84,350 +93,134 @@ public Calculator() {
         inText.setEditable(false);
         inText.setBackground(Color.WHITE);
         inText.setFont(new Font("Comic Sans MS", Font.PLAIN, 33));
+        inText.addKeyListener(new CustomKeyListener(this)); // keylistener for inText
```

Comment:
I think you should apply the listener on the entire app and not only on the inText.